x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_edit/fields/history_window_start/history_window_start_edit_form.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { type FormData, type FormSchema } from '../../../../../../../../shared_imports';
+import { RuleFieldEditFormWrapper } from '../rule_field_edit_form_wrapper';
+import { HistoryWindowStartEditAdapter } from './history_window_start_edit_adapter';
+import type { HistoryWindowStart } from '../../../../../../../../../common/api/detection_engine';
+import { schema } from '../../../../../../../rule_creation_ui/components/step_define_rule/schema';
+import {
+  convertHistorySizeToStart,
+  convertHistoryStartToSize,
+} from '../../../../../../../../common/utils/history_window';
+import { DEFAULT_HISTORY_WINDOW_SIZE } from '../../../../../../../../common/constants';
+
+export function HistoryWindowStartEditForm(): JSX.Element {
+  return (
+    <RuleFieldEditFormWrapper
+      component={HistoryWindowStartEditAdapter}
+      ruleFieldFormSchema={historyWindowFormSchema}
+      deserializer={deserializer}
+      serializer={serializer}
+    />
+  );
+}
+
+interface HistoryWindowFormData {
+  historyWindowSize: HistoryWindowStart;
+}
+
+function deserializer(defaultValue: FormData): HistoryWindowFormData {
+  return {
+    historyWindowSize: defaultValue.history_window_start
+      ? convertHistoryStartToSize(defaultValue.history_window_start)
+      : DEFAULT_HISTORY_WINDOW_SIZE,
+  };
+}
+
+function serializer(formData: FormData): { history_window_start: HistoryWindowStart } {
+  return {
+    history_window_start: convertHistorySizeToStart(formData.historyWindowSize),
+  };
+}
+
+export const historyWindowFormSchema = {
+  historyWindowSize: schema.historyWindowSize,
```

Comment:
From architectural POV it looks better to bring `historyWindowSize` and `newTermsFields` configurations like `label`, `helpText` and etc to the corresponding edit components. When edit components rendered conditionally only for new terms rule types validation doesn't required check for the rule type.

The only thing which needs to be addressed is user input persistence. But it can be easily done via a hook.

Reply:
That's a great idea! I was also considering a way to share schemas but couldn't figure out how. Your approach actually works. I've refactored the code accordingly.